tests/Unit/IO/H5/Test_CombineH5.cpp

```diff
@@ -0,0 +1,226 @@
+    const auto& dat_file = h5_file.get<h5::Dat>(subfile_name);
+    CHECK(dat_file.get_version() == version);
+    CHECK(dat_file.get_legend() == legend);
+    CHECK(dat_file.get_data() == data);
+  }
+}
+
+void test() {
+  const std::string combined_filename{"MightyMorphinPowerRangers.h5"};
+  if (file_system::check_if_file_exists(combined_filename)) {
+    file_system::rm(combined_filename, true);
+  }
+  const std::vector<std::string> individual_filenames{
+      "RedRanger.h5", "BlackRanger.h5", "BlueRanger.h5", "YellowRanger.h5",
+      "PinkRanger.h5"};
+  const std::vector<std::string> subfile_names{"Subfile1", "Subfile2"};
+  // All subfiles can just share the same legend. The data doesn't matter, only
+  // the times for this test.
+  const std::vector<std::string> legend{"Time", "Data"};
+  const std::vector<std::vector<std::vector<double>>> data{
+      // This file doesn't keep the last time
+      {{0.0, 0.0}, {1.0, 1.0}, {2.0, 0.0}},
+      // This file keeps all times
+      {{1.5, 2.0}, {2.1, 3.0}},
+      // This file is completely discarded because the next files' first time is
+      // the same as this first time
+      {{3.0, 0.0}, {4.0, 0.0}, {5.0, 0.0}},
+      // This file only keeps the first time
+      {{3.0, 4.0}, {4.1, 0.0}, {4.5, 0.0}},
+      // This file keeps all times
+      {{4.0, 5.0}, {5.5, 6.0}, {6.0, 7.0}}};
+
+  const Matrix expected_data{{0.0, 0.0}, {1.0, 1.0}, {1.5, 2.0}, {2.1, 3.0},
+                             {3.0, 4.0}, {4.0, 5.0}, {5.5, 6.0}, {6.0, 7.0}};
+
+  // Write the individual files
+  {
+    for (size_t i = 0; i < individual_filenames.size(); i++) {
+      const std::string& filename = individual_filenames[i];
+      if (file_system::check_if_file_exists(filename)) {
+        file_system::rm(filename, true);
+      }
+      h5::H5File<h5::AccessType::ReadWrite> h5_file{filename};
+      for (const std::string& subfile_name : subfile_names) {
+        auto& dat_file = h5_file.insert<h5::Dat>(subfile_name, legend);
+        dat_file.append(data[i]);
+        h5_file.close_current_object();
+      }
+    }
+  }
+
+  // Combine the H5 files
+  h5::combine_h5_dat(individual_filenames, combined_filename,
+                     Verbosity::Verbose);
+
+  REQUIRE(file_system::check_if_file_exists(combined_filename));
+
+  {
+    const h5::H5File<h5::AccessType::ReadOnly> h5_file{combined_filename};
+    for (const std::string& subfile_name : subfile_names) {
+      CAPTURE(subfile_name);
+      const auto& dat_file = h5_file.get<h5::Dat>(subfile_name);
+      const Matrix dat_data = dat_file.get_data();
+      CHECK(expected_data == dat_data);
+      h5_file.close_current_object();
+    }
+  }
+
+  if (file_system::check_if_file_exists(combined_filename)) {
+    file_system::rm(combined_filename, true);
+  }
+
+  for (const std::string& filename : individual_filenames) {
+    if (file_system::check_if_file_exists(filename)) {
+      file_system::rm(filename, true);
+    }
+  }
```

Comment:
It would be good to add tests for catching unordered files passed in, and for unordered data in the subfiles. I worry that these will both be common mistakes.